src/app/course/course-question-snippet/course-question-snippet.component.ts

```diff
@@ -80,4 +93,73 @@ export class CourseQuestionSnippetComponent implements OnInit {
         }
         return '';
     }
+
+    deleteReport(): void {
+        this.questionReportService.deleteReport(this.reportUQJ.report.id).subscribe(() => {
+            this.notificationsService
+                .show('The Report has been Deleted Successfully.', {
+                    status: TuiNotification.Success
+                }).subscribe();
+            this.reportUQJ.report = {} as QuestionReport;
+        });
+    }
+
+    fillReport(): void {
+        this.formGroup.setValue({
+            description: this.reportUQJ.report.report,
+            description_text: this.reportUQJ.report.report_details
+        });
+    }
+
+    createModal(content: unknown, uqj: UQJ): void {
+        this.reportUQJ = uqj;
+        this.formGroup.reset();
+        if (this.reportUQJ.report.id != null) {
+            this.modalService.open(content, {ariaLabelledBy: 'modal-basic-title', centered: true});
```

Comment:
Why not just call the fillmodal method if it is not a null ID. You don't have to repeat line 118

Reply:
Good catch, I fixed this! =